gensim/test/test_nmf.py

```diff
@@ -86,18 +86,22 @@ def test_transform(self):
         transformed = self.model[doc]
 
         vec = matutils.sparse2full(transformed, 2)  # convert to dense vector, for easier equality tests
-        expected = [0., 1.]
+        # The results sometimes differ on Windows, for unknown reasons.
+        # See https://github.com/RaRe-Technologies/gensim/pull/2481#issuecomment-549456750
+        expected = [0.03028875, 0.96971124]
+
         # must contain the same values, up to re-ordering
-        self.assertTrue(np.allclose(sorted(vec), sorted(expected)))
+        self.assertTrue(np.allclose(sorted(vec), sorted(expected), atol=1e-3))
```

Comment:
I think the real problem is here. We're adjusting these tolerances just to make the tests pass, without understanding what's actually going on and why they need to change under some conditions.

Reply:
Yes. What's the alternative?

Reply:
Hey, I quickly checked the reasoning I made there again. I tracked the divergence down to the dot product (https://github.com/anotherbugmaster/gensim/blob/54ceed438a228be0c170599513b5308016f80ce4/gensim/models/nmf.py#L769) as the first location that a different result is created on both platforms. Minor googling' yielded that the difference might be because of the difference between MKL and openBLAS (https://stackoverflow.com/questions/26609475/numpy-performance-differences-between-linux-and-windows) **HOWEVER** when I run  numpy.show_config() on the windows machine used to test this, only openblas is found (I removed Virtualbox, sorry). So sadly, I'm not so sure of the conclusion I made in 2019 anymore.